01-basics.egg

```diff
@@ -108,11 +108,33 @@
 
 ;; **Exercises**
 ;;
-;; 1. Define the distributivity rule for multiplication over addition. Find two expressions that 
-;;    can be proven equivalent using this rule.
-;; 2. Write a rule that asserts `x + 0 = x` and `x * 1` = x. Run egglog over the rules we have above and this rule.
-;;    Try to increase the number of iterations. What happens? Why?
-;; 3. Egglog has the `birewrite` keyword, which is similar to `rewrite`, but it allows rewriting
-;;    in both directions. Make the associativity rule above a birewrite rule. Looking at the rules
+;; 1. Define the distributivity rule for multiplication over addition. When you're done, the
+;;    following code should pass.
+;;     ```
+;;     ; expr3 = 2 * (x + 3)
+;;     (let expr3 (Mul (Num 2) (Add (Var "x") (Num 3))))
+;;     ; expr4 = 6 + 2 * x
+;;     (let expr4 (Add (Num 6) (Mul (Num 2) (Var "x"))))
+;;     (fail (check (= expr3 expr4)))
+;;     (run 10)
+;;     (check (= expr3 expr4))
+;;     ```
+
+(rewrite (Mul a (Add b c)) (Add (Mul a b) (Mul a c)))
+
+; expr3 = 2 * (x + 3)
+(let expr3 (Mul (Num 2) (Add (Var "x") (Num 3))))
+; expr4 = 6 + 2 * x
+(let expr4 (Add (Num 6) (Mul (Num 2) (Var "x"))))
+(fail (check (= expr3 expr4)))
+(run 10)
+(check (= expr3 expr4))
+
+;; 2. Egglog has the `birewrite` keyword, which is similar to `rewrite`, but it allows rewriting
+;;    in both directions. Try making some of the rules above into birewrite rules. Looking at the rules
 ;;    we have defined, when is a birewrite rule useful and when it is not? When is it not 
 ;;    even well-defined?
+
+; A birewrite rule is useful when you would otherwise write out two rewrites.
+; A birewrite rule is not useful when the two side are alpha-equivalent (equivalent up to names).
```

Comment:
call out commutativity as an example. Or in general, add an example to each line here